pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1774,6 +1777,15 @@ public TopicStats getStats(boolean getPreciseBacklog, boolean subscriptionBacklo
 
         stats.ledgers = Lists.newArrayList();
         List<CompletableFuture<String>> futures = includeLedgerMetadata ? Lists.newArrayList() : null;
+        BookKeeperAdmin bookKeeperAdmin = brokerService.pulsar().getBookKeeperAdmin();
+        Collection<BookieId> allBookies;
+        try {
+            allBookies = bookKeeperAdmin.getAllBookies();
```

Comment:
This is mixing a blocking call into an async method. Instead we could be using the MetadataCache to have the list of bookies available.

Reply:
Good point!